storage/reader.go

```diff
@@ -77,9 +80,76 @@ type ReaderObjectAttrs struct {
 //
 // The caller must call Close on the returned Reader when done reading.
 func (o *ObjectHandle) NewReader(ctx context.Context) (*Reader, error) {
+	if o.c.gc != nil {
+		return o.newRangeReaderWithGRPC(ctx, 0, 0)
+	}
 	return o.NewRangeReader(ctx, 0, -1)
 }
 
+// newRangeReaderWithGRPC creates a new Reader with the given range that uses
+// gRPC to read Object content.
+func (o *ObjectHandle) newRangeReaderWithGRPC(ctx context.Context, offset, length int64) (r *Reader, err error) {
+	ctx = trace.StartSpan(ctx, "cloud.google.com/go/storage.Object.newRangeReaderWithGRPC")
+	defer func() { trace.EndSpan(ctx, err) }()
+
+	if o.c.gc == nil {
+		err = fmt.Errorf("handle doesn't have a gRPC client initialized")
+		return
+	}
+	if err = o.validate(); err != nil {
+		return
+	}
+	if length < 0 {
```

Comment:
Negative length should mean read until the end of the object.

Reply:
Ok, I will coerce it to `0` to read to the end instead of returning an error. Supplying a negative value will [result in an error](https://github.com/googleapis/googleapis/blob/master/google/storage/v2/storage.proto#L128-L129) for the grpc API.

Reply:
Coercing length to `0` when it is negative to indicate "read to the end". Done.

Reply:
Yeah, I don't think this is a thing in HTTP either, but it's a convention used in this library-- see the NewRangeReader docs https://pkg.go.dev/cloud.google.com/go/storage#ObjectHandle.NewRangeReader